File: MyFarmProduce.Domain/Entities/Customer.cs
namespace MyFarmProduce.Domain.Entities;

public class Customer
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PasswordHash { get; set; } = string.Empty;
    public bool IsAdmin { get; set; }
    public string? PhotoUrl { get; set; }

    /// <summary>True until an admin-created user changes the default password on first login.</summary>
    public bool MustChangePassword { get; set; }

    public ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: domain/src/main/java/com/myfarmproduce/domain/entity/Customer.java
package com.myfarmproduce.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "customers")
@Getter
@Setter
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 150)
    private String name = "";

    @Column(nullable = false, length = 30)
    private String phone = "";

    @Column(nullable = false, unique = true, length = 256)
    private String email = "";

    @Column(name = "password_hash", nullable = false, length = 512)
    private String passwordHash = "";

    @Column(name = "is_admin", nullable = false)
    private boolean admin;

    @Column(name = "photo_url", length = 500)
    private String photoUrl;

    /** True until an admin-created user changes the default password on first login. */
    @Column(name = "must_change_password", nullable = false)
    private boolean mustChangePassword;

    @OneToMany(mappedBy = "customer")
    private List<Order> orders = new ArrayList<>();
}
